### PR TITLE
windowrules: add fullscreenstate field

### PR DESCRIPTION
**Pull request by sungyoonc**
#### Describe your PR, what does it fix/add?
Add `fullscreenstate` field for window rules v2.

`fullscreenstate: [internal] [client]` matches windows with specific fullscreen state.

For example
- `fullscreenstate:0 1` matches windows that have internal state of 0 and client state of 1.
- `fullscreenstate:* 2` matches windows that have client state of 2 (internal state can be anything).

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)
Accompanying wiki MR: https://github.com/hyprwm/hyprland-wiki/pull/757

#### Is it ready for merging, or does it need work?
Ready for merging.

### CODE REVIEW

File: src/Compositor.cpp [hunk 1]
```diff
@@ -2339,7 +2343,6 @@ void CCompositor::setWindowFullscreenState(const PHLWINDOW PWINDOW, sFullscreenS
     EMIT_HOOK_EVENT("fullscreen", PWINDOW);
 
     PWINDOW->updateDynamicRules();
-    PWINDOW->updateWindowDecos();
```
Comment (MightyPlaza) [inline src/Compositor.cpp:2342]: why was this removed?

Comment (sungyoonc) [reply src/Compositor.cpp:2342]: `updateWindowAnimatedDecorationValues(PWINDOW)` calls `updateWindowDecos()`.

https://github.com/hyprwm/Hyprland/blob/bdb296a83c595f3814f8159d2cffd25c3edfb842/src/Compositor.cpp#L1941

**Comment by vaxerski** [comment]
why not make "any" a `*` instead of `-1`, or allow both?

**Comment by sungyoonc** [comment]
> why not make "any" a `*` instead of `-1`, or allow both?

Made any a `*`.

**Review by vaxerski — APPROVED** [review]

**Comment by vaxerski** [comment]
with this we can nuke fullscreen in a future mr

**Comment by Buttars** [comment]
@vaxerski can we get better documentation around what "internal" and "client" refers to?

**Comment by vaxerski** [comment]
MRs welcome. Internal means what hyprland does. Client means what the client receives

**Comment by Buttars** [comment]
> MRs welcome. Internal means what hyprland does. Client means what the client receives

I'm happy to open a PR(MR?) if you can help me understand the implications. Specifically I have tried every combination of settings of `fullscreenstate` to make it so my brave browser does not go into fullscreen mode on the client yet it goes into that "fullscreen presentation" F11 style mode every time.

**Comment by MightyPlaza** [comment]
> > MRs welcome. Internal means what hyprland does. Client means what the client receives
> 
> I'm happy to open a PR(MR?) if you can help me understand the implications. Specifically I have tried every combination of settings of `fullscreenstate` to make it so my brave browser does not go into fullscreen mode on the client yet it goes into that "fullscreen presentation" F11 style mode every time.
> 
> If this isn't the appropriate place to discuss this let me know how I can reach out.

F11 is controlled by brave, so it will control the client fullscreen mode
You need to use a dispatcher to make it fullscreen (`fullscreenstate 2 0`)
additionally the style can likely be changed

**Comment by Buttars** [comment]
@MightyPlaza  This worked perfectly. My problem is I was trying to apply it as a window rule.

I'll open a PR to update the documentation soon. Thanks!

**Comment by Buttars** [comment]
@MightyPlaza  @vaxerski  thanks kings 👑 👉🏻  https://github.com/hyprwm/hyprland-wiki/pull/892
